### PR TITLE
vim-patch:8.0.1179: Test_popup_and_window_resize() does not always pass

### PR DESCRIPTION
**Pull request by blueyed**
Problem:    Test_popup_and_window_resize() does not always pass.
Solution:   Do not use $VIMPROG, pass the Vim executable in the vimcmd file.
            (Ozaki Kiichi, closes vim/vim#2186)
https://github.com/vim/vim/commit/631820536e4084e01bf990f9314ec385b60b21d7

NOTE: Does not add `GetVimProg()`, but keeps/syncs existing usages of it
in tests (which obviously are all skipped, mostly because of
`!has('terminal')`).  Likely to be replaced with `v:progpath` in Vim [1], or
at least that would/should be used with Neovim then probably.

1: https://github.com/vim/vim/pull/4806

### CODE REVIEW

**Comment by janlazo** [comment]
Add https://github.com/vim/vim/commit/9570aacdb82c1eb5645d0fbc35bf9728d7334b23 here.

**Comment by blueyed** [comment]
Added "patch 8.0.1526: no test using a screen dump yet" also.  Amended `GetVimProg()`.

**Comment by blueyed** [comment]
Note: unlike on Vim (https://github.com/vim/vim/pull/4806#issuecomment-520623678), `v:progpath` is absolute always (https://github.com/blueyed/neovim/blob/8849011fbe10c8f348a54fbc6167a7625a7ce733/test/functional/options/defaults_spec.lua#L195-L198).

File: src/nvim/testdir/shared.vim [hunk 1]
```diff
@@ -196,16 +196,26 @@ func s:feedkeys(timer)
   call feedkeys('x', 'nt')
 endfunc
 
+" Get $VIMPROG to run Vim executable.
+" The Makefile writes it as the first line in the "vimcmd" file.
+" Nvim: uses $NVIM_TEST_ARG0.
+func GetVimProg()
+  if empty($NVIM_TEST_ARG0)
+    " Assume the script was sourced instead of running "make".
+    return '../../../build/bin/nvim'
```
Comment (justinmk) [inline src/nvim/testdir/shared.vim]: could use v:progpath then

**Review by justinmk — APPROVED** [review]

File: src/nvim/testdir/shared.vim (hunk 1 above)
Comment (blueyed) [reply src/nvim/testdir/shared.vim]: This tries to follow the (not really clear) use case mentioned in https://github.com/vim/vim/pull/4806#issuecomment-520847278.

**Pull request merged**